File: advert-service/internal/service/video.service.ts
import { IVideoRepo } from "#internal/interfaces/repo/video.repo.interface.js";
import { IVideoService } from "#internal/interfaces/service/video.service.interface.js";

export class VideoService implements IVideoService{
    constructor(private readonly videoRepo: IVideoRepo) {}

    async createVideo(data: any): Promise<any> {
        return await this.videoRepo.create(data)
    }
}

File: products-service/src/main/java/com/matreshka/products_service/application/VideoService.java
package com.matreshka.products_service.application;

import com.matreshka.products_service.application.port.IVideoService;
import com.matreshka.products_service.delivery.broker.dto.VideoEvent;
import com.matreshka.products_service.internal.infrastructure.persistence.VideoEntity;
import com.matreshka.products_service.internal.infrastructure.persistence.mapper.IVideoMapper;
import com.matreshka.products_service.internal.repo.IVideoRepo;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class VideoService implements IVideoService {

    private final IVideoRepo videoRepo;
    private final IVideoMapper videoMapper;

    public void processVideo(VideoEvent event) {
        VideoEntity video = videoMapper.toVideoEntity(event);
        videoRepo.save(video);
    }
}